### PR TITLE
feat(thirdparty): sign in with apple

### LINKED ISSUE #616: Add "Sign in with Apple"

### PR DESCRIPTION
**Pull request by lfleischmann**
# Description

This PR adds support for `Sign in with Apple`.

Closes #616 

# Implementation

- Add an Apple provider implementation (f51b68d7217f0e51b84be4883a5717ea4a7bb6a8, a4ea3a80f5e071d5b8c9fb7328f23bcd9148a96d, 793971365655f78ba02f55130e9f121a1634a3a9)
- Add a POST `thirdparty/callback` handler that redirects to the GET handler (9b6855d7463677932dd83c5fb9713e621ad9453d, 1def5717c7c2b0a4d8d957d05d4225ca11fa7699)
  - This is required because Apple enforces a form POST response when requesting scopes
- Add the Apple icon to `elements` for display in `Sign in with ...` button (11912dfd1020f14727ccd60b664fa7472185bf45)
- Add a `siwa` command that allows generating client secrets as per [Apple requirements](https://developer.apple.com/documentation/sign_in_with_apple/generate_and_validate_tokens#3262048) (e87b0ece413ee0fd6e6222d5e68bd5da5f20fc77)
- Document `Sign in with apple` in social guide (544d80830807cfa5d81f13d9618f898abb7816c9)

# Tests

Thirdparty handler tests have been extended accordingly.

### CODE REVIEW

File: docs/docs/guides/social/apple.mdx [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Login with Apple
+
+To enable Apple login for your project, you need to
+- Create a `Sign In With Apple`-capable application with Apple.
+- Create and your Apple credentials and configure them with Hanko.
```
Comment (bjoern-m) [inline docs/docs/guides/social/apple.mdx:19]:
```suggestion
- Create your Apple credentials and configure them with Hanko.
```

File: docs/docs/guides/social/apple.mdx [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+  Per default the key file name should look like this: `AuthKey_XXXXXXXXXX.p8`. Because we use Docker in this case, we
+  will use the path to the key file on your filesystem to mount the file into the Docker container (we'll get to this,
+  see the command below).
+- `team_id`: This is the Team ID your Apple Developer account is associated with.
+  - See [Locate your Team ID](https://developer.apple.com/help/account/manage-your-team/locate-your-team-id/#:~:text=In%20the%20top%20navigation%20of,to%20view%20your%20Team%20ID.)
+    in the official Apple documentation.
+- `services_id`: This is `Services ID` you created in the [Register a Services ID](#register-a-services-id) step. You
+  can always review this ID in the Apple Developer console:
+  - On [Certificates, Identifiers & Profiles](https://developer.apple.com/account/resources), click `Identifiers` in the sidebar,
+  then select `Services IDs` in the dropdown in the right.
```
Comment (bjoern-m) [inline docs/docs/guides/social/apple.mdx:149]:
```suggestion
  then select `Services IDs` from the dropdown menu on the right.
```

File: docs/docs/guides/social/apple.mdx [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+<TabItem value="self-hosted" label="Self-hosted">
+```
+
+When self-hosting you need to adjust your backend
+[configuration file](https://github.com/teamhanko/hanko/blob/main/backend/config/config.yaml). You must provide
+the following data:
+  - your provider [`redirect_url`](#redirect-url)
+  - your `client ID`, which in this case is your [`Services ID`](#configure-service)
+  - your client [`secret`](#generate-secret)
+  - an`error_redirect_url`: This is a URL in your frontend the Hanko API redirects to if an error occurs during third
```
Comment (like-a-bause) [inline docs/docs/guides/social/apple.mdx:224]:
```suggestion
  - an `error_redirect_url`: This is a URL in your frontend the Hanko API redirects to if an error occurs during third
```

**Review by like-a-bause — CHANGES_REQUESTED** [review]
Works as intended. I could follow the guide and get it working.

File: docs/docs/guides/social/apple.mdx [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+docker run -v <key_file_on_host>:<key_file_in_container> ghcr.io/teamhanko/hanko siwa \
+--private_key <key_file_in_container>  \
+--team_id <your_team_id> \
+--services_id <your_services_id> \
+--key_id <your_key_id> > client_secret.txt
+```
+
+The command will save the client secret in a file `client_secret.txt`.
+
+:::caution
```
[the remark this answers is not in the document]
Comment (like-a-bause) [reply docs/docs/guides/social/apple.mdx:173]: I think a Fire would escalated things a little.

**Pull request merged**